src/main.go

```diff
@@ -56,6 +56,24 @@ func getCname(path string) (string, string, error) {
 	return cname, version, nil
 }
 
+func checkEFI(expected_loader_entry string) error {
+	_, err := os.Stat("sys/firmware/efi")
+	if err != nil {
+		return errors.New("not EFI booted")
+	}
+
+	data, err := os.ReadFile("/sys/firmware/efi/efivars/LoaderEntrySelected-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f")
+	if err != nil {
+		return errors.New("not booted with systemd EFI stub")
+	}
+
+	if string(data[4:]) == expected_loader_entry {
+		return nil
```

Comment:
imo it would be more idiomatic go if you'd check for non-equal, throw the error and have the return nil as the last statement of the function outside of the if. YMMV.